node/src/components/consensus/highway_core/state.rs

```diff
@@ -339,7 +340,8 @@ impl<C: Context> State<C> {
     /// children of the previously selected block (or from all blocks at height 0), until a block
     /// is reached that has no children with any votes.
     pub(crate) fn fork_choice<'a>(&'a self, pan: &Panorama<C>) -> Option<&'a C::Hash> {
-        let start_time = Timestamp::now();
+        let clock = Clock::new();
```

Comment:
Not sure if that's expensive, but in `reactor.rs`, we seem to create a new clock only once, and store it in a field.

Reply:
I wanted to keep all modifications to the minimum without changing much of the present code.

Reply:
Are you sure this takes a negligible amount of time? [The docs at least say that `new` does some checks](https://docs.rs/quanta/0.6.5/quanta/struct.Clock.html#method.new).

Reply:
Oh I see your point, I can add it as a field

Reply:
Added it as a field to State